target/riscv/cpu_helper.c

```diff
@@ -622,15 +631,24 @@ static int get_physical_address(CPURISCVState *env, hwaddr *physical,
             /* Fetch access check failed */
             return TRANSLATE_FAIL;
 #if defined(TARGET_CHERI) && !defined(TARGET_RISCV32)
-        } else if (access_type == MMU_DATA_CAP_STORE && !(pte & PTE_SC)) {
-            /* SC inhibited */
+        } else if (access_type == MMU_DATA_CAP_STORE &&
+                   !(pte & PTE_SC) && !(pte & PTE_SCM)) {
```

Comment:
Another way you could write this is:

```
(pte & (PTE_SC | PTE_SCM) == 0)
```

I find that this style can be a bit clearer when doing things like:

```
(pte & (PTE_SC | PTE_SCM) == PTE_SC)
```

but I don't have a strong opinion.